src/tip/tip_meta_finder.py

```diff
@@ -2,18 +2,28 @@
 from importlib.abc import MetaPathFinder
 from importlib.util import spec_from_file_location
 
+from tip import cache
+
 
 class TipMetaFinder(MetaPathFinder):
     """Finder which accepts a list of packages in custom directories to import them from there."""
 
     def __init__(self, packages_to_mount):
         self.packages_to_mount = packages_to_mount
+        if cache.is_enabled():
+            self.packages_to_mount_cache = {}
 
     def find_spec(self, fullname, path, target=None):
         # pylint: disable=unused-argument
         path = []
         if fullname in self.packages_to_mount:
-            path.append(self.packages_to_mount[fullname])
+            try:
+                path.append(self.packages_to_mount_cache[fullname])
+            except KeyError:
+                self.packages_to_mount_cache[fullname] = cache.add(self.packages_to_mount[fullname])
```

Comment:
🤦

Reply:
Your cache isn't a cache. It just copies all files everytime you're making imports (what is imported is already cached by python in RAM). I'm surprised that you noticed some speed up.

Reply:
It doesn't copy if package is cached.